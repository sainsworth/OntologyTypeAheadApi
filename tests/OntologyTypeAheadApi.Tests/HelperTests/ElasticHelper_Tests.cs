using Newtonsoft.Json;
using OntologyTypeAheadApi.Models.Elastic;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Microsoft.VisualStudio.TestTools.UnitTesting;
using OntologyTypeAheadApi.Helpers;
using Shouldly;

namespace OntologyTypeAheadApi.Tests.HelperTests
{
    [TestClass]
    public class  ElasticHelper_Tests
    {
        private static Dictionary<string, string> test_items = new Dictionary<string, string>()
        {
            { "One", "Item One" },
            { "Two", "Item two" }
        };

        private static string accessor = "Accessor";

        [TestMethod]
        public void ElasticHelper_GetBulkJsonForItem()
        {
            var result = ElasticHelper.GetBulkJsonForItem(accessor, "One", "Item One");

            result.ShouldMatchApproved(x =>
            {
                x.SubFolder("Approvals");
            });
        }

        [TestMethod]
        public void ElasticHelper_GetBulkJsonForItems()
        {
            var result = ElasticHelper.GetBulkJsonForItems(accessor, test_items);

            result.ShouldMatchApproved(x =>
            {
                x.SubFolder("Approvals");
            });
        }

        [TestMethod]
        public void ElasticHelper_GetQueryJson_All()
        {
            ElasticHelper.GetQueryJson(10, 2, Enums.QueryType.All, "").ShouldMatchApproved(x =>
            {
                x.SubFolder("Approvals");
            });
        }

        [TestMethod]
        public void ElasticHelper_GetQueryJson_Contains()
        {
            ElasticHelper.GetQueryJson(10, 2, Enums.QueryType.Contains, "Contains").ShouldMatchApproved(x =>
            {
                x.SubFolder("Approvals");
            });
        }

        [TestMethod]
        public void ElasticHelper_GetQueryJson_StartsWith()
        {
            ElasticHelper.GetQueryJson(10, 2, Enums.QueryType.StartsWith, "StartsWith").ShouldMatchApproved(x =>
            {
                x.SubFolder("Approvals");
            });
        }

        [TestMethod]
        public void ElasticHelper_GetQueryJson_Exact()
        {
            ElasticHelper.GetQueryJson(10, 2, Enums.QueryType.Exact, "Exact").ShouldMatchApproved(x =>
            {
                x.SubFolder("Approvals");
            });
        }
    }
}
