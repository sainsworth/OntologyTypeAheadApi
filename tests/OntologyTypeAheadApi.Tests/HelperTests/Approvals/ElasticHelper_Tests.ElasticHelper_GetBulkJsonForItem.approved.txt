{"create":{"_index":"accessor","_type":"accessor","_id":"One"}}
{"doc":{"Label":"Item One"}}
